base/logger.go

```diff
@@ -34,3 +37,34 @@ func logCollationWorker(collateBuffer chan string, logger *log.Logger, bufferSiz
 		}
 	}
 }
+
+//runLogDeletion will delete rotated logs for the supplied logLevel. It will only perform these deletions when the
+//cumulative size of the logs are above the supplied sizeLimitMB.
+//logDirectory is the supplied directory where the logs are stored.
+func runLogDeletion(logDirectory string, logLevel string, sizeLimitMB int) {
```

Comment:
It would be nicer if this function returned errors, rather than logged them.
Think about how you'd test this function under error scenarios.

We can always log any errors returned in the caller goroutine.